### DISCUSSION TITLE
Customize display name?

### DISCUSSION THREAD

**Discussion by KamilPacanek** [Yvand/EntraCP#301] [General]
I see that display name for Entra groups is prefixed with a `(Role)`. Can it be somehow done the same for the regular users? Best if that prefix can be added based on some other condition (ex. if expression based on other properties resolves to true/false).

Right now I can only see that it will require some coding to extent EntraCP with that functionality on my own. Do you see some downsides/caveats of doing such thing?

**Comment by Yvand** [comment]
@KamilPacanek it is not documented, but your can configure a static leading display text by setting the property `EntityDisplayTextPrefix `:

```powershell
Add-Type -AssemblyName "Yvand.EntraCP, Version=1.0.0.0, Culture=neutral, PublicKeyToken=65dc6b5903b51636"
$config = [Yvand.EntraClaimsProvider.EntraCP]::GetConfiguration()
$settings = $config.Settings
$settings.EntityDisplayTextPrefix = "YourPrefixValue"
$config.ApplySettings($settings, $true)
```

**Reply by KamilPacanek** [reply]
Ah thank you.

**Reply by KamilPacanek** [reply]
@Yvand and how complicated would that be if I would like to extend solution to append some prefix dynamically to the display name? So it returns in that "extended" display name on people pickers?

Edit: ah now that comes to my mind - I will have a lot of issues when doing that, because I have already deployed on environment your original one. And when I do modification I can't sign the assemblies using your key (obviously). So my assemblies will have different key - which breaks all EntraCP and Identity Provider configurations - I have already went through that once.
Am I missing something and it can be done differently?

**Reply by Yvand** [reply]
@KamilPacanek yes, you would need to create your own DLL, and yes it has consequences in terms of maintenance, and I strongly discourage you to go this road, unless you really need to.

But since you would essentially create a new claims provider, all it would take to switch to yours is to update the property `ClaimProviderName` in `Get-SPTrustedIdentityTokenIssuer`.

The real, huge pain point, is the management of the dependencies
